spec-dtslint/operators/combineLatest-spec.ts

```diff
@@ -0,0 +1,110 @@
+      const res = a.pipe(combineLatest(b, c, d, e)); // $ExpectType Observable<[number, string, string, string, string]>
+    });
+
+    it('should infer correctly with 5 params', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of('d', 'e', 'f');
+      const d = of('g', 'h', 'i');
+      const e = of('j', 'k', 'l');
+      const f = of('m', 'n', 'o');
+      const res = a.pipe(combineLatest(b, c, d, e, f)); // $ExpectType Observable<[number, string, string, string, string, string]>
+    });
+
+    it('should only accept maximum params of 5', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of('d', 'e', 'f');
+      const d = of('g', 'h', 'i');
+      const e = of('j', 'k', 'l');
+      const f = of('m', 'n', 'o');
+      const g = of('p', 'q', 'r');
+      const res = a.pipe(combineLatest(b, c, d, e, f, g)); // $ExpectError
+    });
+  });
+
+  describe('with project parameter', () => {
+    it('should infer correctly with project param', () => {
+      const a = of(1, 2, 3);
+      const res = a.pipe(combineLatest(v1 => 'b')); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 1 param', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const res = a.pipe(combineLatest(b, (a, b) => b)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 2 params', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of('d', 'e', 'f');
+      const res = a.pipe(combineLatest(b, c, (a, b, c) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 3 params', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of('d', 'e', 'f');
+      const d = of('g', 'h', 'i');
+      const ref = a.pipe(combineLatest(b, c, d, (a, b, c, d) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 4 params', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of('d', 'e', 'f');
+      const d = of('g', 'h', 'i');
+      const e = of('j', 'k', 'l');
+      const res = a.pipe(combineLatest(b, c, d, e, (a, b, c, d, e) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 5 params', () => {
+      const a = of(1, 2, 3);
+      const b = of('a', 'b', 'c');
+      const c = of('d', 'e', 'f');
+      const d = of('g', 'h', 'i');
+      const e = of('j', 'k', 'l');
+      const f = of('m', 'n', 'o');
+      const res = a.pipe(combineLatest(b, c, d, e, f, (a, b, c, d, e, f) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with array param', () => {
+      const a = of(1, 2, 3);
+      const b = [of('a', 'b', 'c')];
+      const res = a.pipe(combineLatest(b, (a, b) => b)); // $ExpectType Observable<Observable<string>>
```

Comment:
@cartant - This is the one I was asking you about.
I would expect the return would be `Observable<[number, string, string]>`
Instead, the typing when I hover over res was `Observable<Observable<string>>`.
However, when I put this into Stackblitz and console log the result it does return me what I expected.
Please advise.

Reply:
The problem is that the typings of the `combineLatest` operator and the `combineLatest` creator function don't match the implementation's behaviour. If a single array is passed, it's assumed that the array contains the observables that are to be combined - that is, the array itself is not treated as an `ObservableInput`.

From here, I think you should comment out this test and add a TODO that references this comment. And, if you are keen, create an issue that references both. I will look at fixing this later.

Thanks for the work you've put into writing the tests.